zerver/views/custom_profile_fields.py

```diff
@@ -116,6 +117,25 @@ def reorder_realm_custom_profile_fields(request: HttpRequest, user_profile: User
     try_reorder_realm_custom_profile_fields(user_profile.realm, order)
     return json_success()
 
+@human_users_only
+@has_request_variables
+def remove_user_custom_profile_data(request: HttpRequest, user_profile: UserProfile,
+                                    data: List[int]=REQ(validator=check_list(
+                                                        check_int))) -> HttpResponse:
+    for field_id in data:
+        try:
+            field = CustomProfileField.objects.get(realm=user_profile.realm, id=field_id)
+            field_value = CustomProfileFieldValue.objects.get(field=field, user_profile=user_profile)
+            field_value.delete()
+            payload = dict(user_id=user_profile.id, custom_profile_field=dict(id=field_id,
+                                                                              value=None))
+            notify_user_update_custom_profile_data(payload, user_profile.realm.id)
+        except CustomProfileField.DoesNotExist:
+            return json_error(_('Value for field id {id} not exist.').format(id=field_id))
```

Comment:
For `DELETE`, I think you don't want to throw an exception here; DELETE is required to be idempotent.

Reply:
got it. change a little on how it throws an exception if CustomProfileField is does not exist it throws an error but in case of CustomProfileFieldValue it pass. Does this make sense?

Reply:
Ahh, yes, that makes sense.